src/cloudsc_pyiface/CMakeLists.txt

```diff
@@ -0,0 +1,158 @@
+# (C) Copyright 1988- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+# Define this dwarf variant as an ECBuild feature
+ecbuild_add_option( FEATURE CLOUDSC_FORTRAN_PYIFACE
+    DESCRIPTION "Build the Python interface to the (slightly customized) cloudsc-fortran" DEFAULT ON
+    CONDITION HDF5_FOUND
+)
+ecbuild_add_option( FEATURE CLOUDSC_FORTRAN_PYIFACE_BINARY
+	DESCRIPTION "Build the debug-oriented binary for Python interface of (customized) cloudsc-fortran" DEFAULT OFF
+    CONDITION HAVE_CLOUDSC_FORTRAN_PYIFACE 
+)
+
+if( HAVE_CLOUDSC_FORTRAN_PYIFACE )
+
+    # Utilities to manage Python virtual environments
+    include( python_venv )
+
+    # Set up a custom venv for this variant and install the necessary dependencies
+    set( loki_VENV_PATH ${CMAKE_BINARY_DIR}/venv_pyiface )
+    setup_python_venv( ${loki_VENV_PATH} )
+
+    if( NOT Python3_EXECUTABLE )
+        ecbuild_error("[PyIface] Could not find Python3 executable in virtualenv")
+    endif()
+
+    # Install the f90wrap build dependency and via latest pip
+    execute_process(COMMAND ${Python3_EXECUTABLE} -m pip install --upgrade pip)
```

Comment:
Yes. The reason is that the actual wrapper-generation/build process is unfortunately not cleanly dependency-driven, but instead a set of commands executed in sequence whenever that lib-object is built. Originally, that even included the venv-creation and install step, but it would mean that all this setup stuff gets repeated every time the lib-target was built, whereas it now only re-does wrapper-compile steps upon re-build. 

This can for sure be fixed via better `add_custom_command` that actually captures the respective dependencies, but I think that can be done in a follow-up step. As a first step for this, we would need to define a proper binary target, instead of dangling everything from a single lib-target via `POST_BUILD`, which ignores dependencies, AFAICT.

FWIW, I'm also not sure if the explicit pip upgrade is still needed, but again, happy to keep this for now and clean up later.

Reply:
Thanks, makes sense. I think we don't need to over-engineer it for CLOUDSC, so fine with this, but may be of interest to further refine the wrapper recipe, of course. All of this is doesn't need to be addressed here, though.